[package] Bring back save_source_file

zhxchen17 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#59962 [package] Bring back save_source_file**

This reverts commit 44b021d21b5681c105529881bdbaefb6d3e335f6. Issue: #59737

Differential Revision: [D29113224](https://our.internmc.facebook.com/intern/diff/D29113224)

zhxchen17 committed:
[package] Bring back save_source_file

This reverts commit 44b021d21b5681c105529881bdbaefb6d3e335f6.

[ghstack-poisoned]

zhxchen17 committed:
Update on "[package] Bring back save_source_file"


This reverts commit 44b021d21b5681c105529881bdbaefb6d3e335f6.

[ghstack-poisoned]

zhxchen17 commented:
@zhxchen17 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29113224).

zhxchen17 committed:
Update on "[package] Bring back save_source_file"


This reverts commit 44b021d21b5681c105529881bdbaefb6d3e335f6.

Differential Revision: [D29113224](https://our.internmc.facebook.com/intern/diff/D29113224)

[ghstack-poisoned]

zhxchen17 commented:
@zhxchen17 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29113224).

zhxchen17 committed:
Update on "[package] Bring back save_source_file"


This reverts commit 44b021d21b5681c105529881bdbaefb6d3e335f6.

Differential Revision: [D29113224](https://our.internmc.facebook.com/intern/diff/D29113224)

[ghstack-poisoned]

zhxchen17 commented:
@zhxchen17 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29113224).

SplitInfinity commented on torch/package/package_exporter.py:
```diff
@@ -223,6 +223,60 @@ def __init__(
+    def save_source_file(
+        self, module_name: str, file_or_directory: str, dependencies=True
+    ):
+        """Adds the local file system `file_or_directory` to the source package to provide the code
```
```suggestion
        """Adds the local file system ``file_or_directory`` to the source package to provide the code
```

SplitInfinity commented on torch/package/package_exporter.py:
```diff
@@ -223,6 +223,60 @@ def __init__(
+        self, module_name: str, file_or_directory: str, dependencies=True
+    ):
+        """Adds the local file system `file_or_directory` to the source package to provide the code
+        for `module_name`.
```
```suggestion
        for ``module_name``.
```

SplitInfinity commented on torch/package/package_exporter.py:
```diff
@@ -223,6 +223,60 @@ def __init__(
+        for `module_name`.
+
+        Args:
+            module_name (str): e.g. `my_package.my_subpackage`, code will be saved to provide code for this package.
+            file_or_directory (str): the path to a file or directory of code. When a directory, all python files in the directory
+                are recursively copied using :meth:`save_source_file`. If a file is named "/__init__.py" the code is treated
+                as a package.
+            dependencies (bool, optional): If ``True``, we scan the source for dependencies.
```
```suggestion
            module_name (str): e.g. ``"my_package.my_subpackage"``, code will be saved to provide code for this package.
            file_or_directory (str): the path to a file or directory of code. When a directory, all python files in the directory
                are recursively copied using :meth:`save_source_file`. If a file is named ``"/__init__.py"`` the code is treated
                as a package.
            dependencies (bool, optional): If ``True``, we scan the source for dependencies.
```

SplitInfinity reviewed (approved):
Just a few comments to make sure the docstrings render well in the documentation.

Thanks for working on this!

zhxchen17 committed:
Update on "[package] Bring back save_source_file"


This reverts commit 44b021d21b5681c105529881bdbaefb6d3e335f6. Issue: #59737

Differential Revision: [D29113224](https://our.internmc.facebook.com/intern/diff/D29113224)

[ghstack-poisoned]

zhxchen17 commented:
@zhxchen17 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29113224).
